firestore.rules

```diff
@@ -56,37 +63,49 @@ service cloud.firestore {
       match /tasks/{anyTask} {
         allow read: if request.auth != null;
         // Allow create: if authenticated
-        // DONT EXPORT THE NEXT SENTENCE TO A FUNCTION - FIREBASE HAS A BUG THAT DOESNT ALLOW THAT TO WORK -
-        // KEEP IT LIKE SO
         // Uncommment the canCreateTask will not allow users who already have tasks to create new tasks
-        allow create: if request.auth != null // && ((get(/databases/$(database)/documents/projects/$(anyProject)).data.canCreateTask == true)
+        allow create: if request.auth != null && isUserInvited(anyProject);
         // Uncomment the following to allow unassigned task to be assigned by all
-        allow update: if request.auth != null && (isSpecificProjectAdmin() || isGuide() || isUserAssigneeOrCreator() || isTaskUnassigned() || isUserOnlySendingListeners() || isSuperAdmin());
-        allow delete: if request.auth != null && (isSpecificProjectAdmin() || isCreator() || isSuperAdmin());
+        allow update: if request.auth != null && isSpecificProjectAdmin(anyProject) || (isUserInvited(anyProject) && (isGuide() || isUserAssigneeOrCreator() || isTaskUnassigned() || isUserOnlySendingListeners())) || isSuperAdmin();
+        allow delete: if request.auth != null && isSpecificProjectAdmin(anyProject) || (isUserInvited(anyProject) && isCreator()) || isSuperAdmin();
       }
 
       match /comments/{anyComment} {
         allow read: if request.auth != null;
-        allow create: if request.auth != null;
+        allow create: if request.auth != null && isUserInvited(anyProject) || isSpecificProjectAdmin(anyProject) || isSuperAdmin();
         allow update: if request.auth != null && isCreator();
         allow delete: if request.auth != null && isCreator();
       }
 
+      match /invitation_lists/{anyInvitationList} {
+        allow create: if request.auth != null && (isSpecificProjectAdmin(anyProject) || isSuperAdmin());
+        allow read: if request.auth != null && (isSpecificProjectAdmin(anyProject) || isSuperAdmin());
+        allow update: if request.auth != null && (isSpecificProjectAdmin(anyProject) || isSuperAdmin());
+        allow delete: if request.auth != null && (isSpecificProjectAdmin(anyProject) || isSuperAdmin());
+     }
+
+     match /invitations/{anyInvitation} {
+        allow create: if request.auth != null && (isSpecificProjectAdmin(anyProject) || isSuperAdmin());
```

Comment:
We should check also if the user has permission to create using the invitation_list permissions